cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/plugin/TestMacroEvaluator.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.app.runtime.plugin;
+
+import co.cask.cdap.api.macro.InvalidMacroException;
+import co.cask.cdap.api.macro.MacroEvaluator;
+import com.google.common.collect.ImmutableMap;
+
+import java.util.Map;
+
+public class TestMacroEvaluator implements MacroEvaluator {
+
+  private final Map<String, String> propertySubstitutions;
+  private final Map<String, String> testFunctionSubstitutions;
+
+  public TestMacroEvaluator() {
+    this.propertySubstitutions = ImmutableMap.<String, String>builder()
+      // property-specific tests
+      .put("notype", "Property Macro")
```

Comment:
rather than hardcoding these in the test evaluator, take the maps in as arguments. That way you can create different evaluators depending on what you are trying to test, which will be more readable than looking at these big maps.